File: packages/emitter/testcases/common/expected/types/expected-type-threading/ReturnInControlFlow.cs
namespace TestCases.common.types.expectedtypethreading
{
    [global::Tsonic.Internal.ModuleContainerAttribute]
    public static class ReturnInControlFlow
    {
        public static int getInIf(bool condition)
        {
            if (condition)
            {
                return (int)100;
            }
            return (int)200;
        }

        public static int getInElse(bool condition)
        {
            if (condition)
            {
                return (int)10;
            }
            else
            {
                return (int)20;
            }
        }

        public static int getInWhile(int count)
        {
            while (count > (int)0)
            {
                return (int)50;
            }
            return (int)0;
        }

        public static int getInSwitch(int key)
        {
            switch (key)
            {
                case 1:
                    return (int)100;
                case 2:
                    return (int)200;
                default:
                    return (int)0;
            }
        }
    }
}

File: packages/targets/csharp/emitter/testcases/common/types/expected-type-threading/ReturnInControlFlow.ts
import { int } from "@tsonic/core/types.js";

// Return in if branch should get function's return type
export function getInIf(condition: boolean): int {
  if (condition) {
    return 100;
  }
  return 200;
}

// Return in else branch
export function getInElse(condition: boolean): int {
  if (condition) {
    return 10;
  } else {
    return 20;
  }
}

// Return in while loop
export function getInWhile(count: int): int {
  while (count > 0) {
    return 50;
  }
  return 0;
}

// Return in switch case
export function getInSwitch(key: int): int {
  switch (key) {
    case 1:
      return 100;
    case 2:
      return 200;
    default:
      return 0;
  }
}
